mypy/plugins/functools.py

```diff
@@ -0,0 +1,78 @@
+"""Plugin for supporting the functools standard library module."""
+from typing import Dict, Optional
+from typing_extensions import Final
+
+import mypy.plugin
+from mypy.nodes import ARG_POS, ARG_STAR2, Argument, FuncItem, Var
+from mypy.plugins.common import add_method
+from mypy.types import CallableType, Type
+
+
+functools_total_ordering_makers: Final = {
+    'functools.total_ordering',
+}
+
+_ORDERING_METHODS: Final = {
+    '__lt__',
+    '__le__',
+    '__gt__',
+    '__ge__',
+}
+
+
+def functools_total_ordering_maker_callback(ctx: mypy.plugin.ClassDefContext,
+                                            auto_attribs_default: bool = False) -> None:
+    """Add dunder methods to classes decorated with functools.total_ordering."""
+    if ctx.api.options.python_version < (3, 2):
+        ctx.api.fail("functools.total_ordering is not supported in Python 2 or 3.1", ctx.reason)
+        return
+
+    comparison_methods = _analyze_class(ctx)
+    if not comparison_methods:
+        ctx.api.fail('must define at least one ordering operation: < > <= >=', ctx.reason)
+        return
+
+    root = max(comparison_methods)  # prefer __lt__ to __le__ to __gt__ to __ge__
+    root_method = comparison_methods[root]
+    other_type = _find_other_type(root_method)
+    if isinstance(root_method.type, CallableType):
+        ret_type = root_method.type.ret_type
+    else:
+        ret_type = ctx.api.named_type('__builtins__.bool')
```

Comment:
Shouldn't it just be an error if `__lt__` (etc.) is not callable?

Reply:
If `__lt__` is not callable, it's okay to either generate an error or to fall back to some default behavior.